src/atopile/_shim.py

```diff
@@ -87,72 +87,37 @@ def handle_duplicate(
         return False
 
 
-class Component(L.Module):
+class has_ato_cmp_attrs(L.Module.TraitT.decless()):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self.pinmap = {}
 
-    @L.rt_field
-    def attach_to_footprint(self):
-        return F.can_attach_to_footprint_via_pinmap(self.pinmap)
-
-    @L.rt_field
-    def has_designator_prefix(self):
-        return F.has_designator_prefix_defined(F.has_designator_prefix.Prefix.U)
+    def on_obj_set(self):
+        self.module = cast_assert(L.Module, self.obj)
```

Comment:
there is shortcut, self.get_obj(L.Module)

Reply:
Thanks! Will update in next iteration